api/app/controllers/review.controller.ts

```diff
@@ -444,6 +445,75 @@ export const getAllReviewsForAdmin = async (req: Request, res: Response) => {
   }
 };
 
+export const getReviewsByPaymentStatus = async (
+  req: Request,
+  res: Response
+) => {
+  const status = (req.query.status as string)?.toLowerCase();
+
+  if (!status) {
+    return res.status(400).json({
+      message: "status of 'paid' or 'unpaid' is required",
+    });
+  }
+
+  const validStatuses = ["paid", "unpaid"];
+
+  if (!validStatuses.includes(status)) {
+    return res.status(400).json({
+      message: "status must be either 'paid' or 'unpaid'",
+    });
+  }
+
+  try {
+    const mergedReviews = await Review.findAll({
+      where: { mergedAt: { [Op.not]: null } },
+    });
+    const allCreditTransactions = await Transaction.findAll({
+      where: { transactionType: TRANSACTION_TYPE.CREDIT },
+    });
+
+    const creditTransactionReviewIds = allCreditTransactions.map(
+      (transaction) => transaction.reviewId
+    );
+    const unpaidMergedReviews = mergedReviews.filter(
+      (review) => !creditTransactionReviewIds.includes(review.id)
+    );
+    const paidMergedReviews = mergedReviews.filter((review) =>
+      creditTransactionReviewIds.includes(review.id)
+    );
+
```

Comment:
I think to optimise our query we can fetch all reviews and transactions instead of making multiple calls to the db, and we can modify the queries to fetch only the necessary data. we can use a sequelize include operation to combine the reviews and transactions tables, and then filter the results based on the payment status.

Reply:
I'm not sure if I get your point. But based on our current db schema, I don't think it's possible to include the transaction data when fetching reviews since review doesn't have transactions as a foreign key. We can only fetch transactions and include review but this will only onclude credit transactions. We won't be able to fetch merged but not credited reviews. Hence the two API calls.
Also, we only create a credit transaction when the review has been merged, so I'm not sure how you'd get all the transaction data for reviews that the webhook failed except this way I implemented it. If the webhook fails no transaction is created.